include/hip/hcc_detail/hiprtc.h

```diff
@@ -102,8 +102,9 @@ namespace hip_impl
         {
             if (!result) return HIPRTC_ERROR_INVALID_INPUT;
 
-            *result = hip_impl::demangle(typeid(T).name());
-
+            char * res= hip_impl::demangle(typeid(T).name());
+            result->assign(res == nullptr ? std::string() : res);
+            std::free(res);
```

Comment:
by have different types in the select ternary ops, it in facts converts 'res' into std::string first and assign that instead of char *. How about, replace std::string() with "".

Reply:
`free` would segfault if `res==nullptr`.

Reply:
Sure i will change that to "". I overlooked the ternary operator trap

Reply:
> free would segfault if res==nullptr.

Nope, all will be ok.